src/Farmer/Arm/ResourceGroup.fs

```diff
@@ -39,11 +39,11 @@ type ResourceGroupDeployment =
     interface IParameters with 
         member this.SecureParameters = this.Parameters
     interface IArmResource with
-        member this.ResourceId = resourceGroupDeployment.resourceId this.Name
+        member this.ResourceId = resourceGroupDeployment.resourceId this.Name.Value
```

Comment:
should this be `.Eval()`?
It be useful to have a `ResourceName` property which is `this.Name.Eval()` and can be reused everywhere it is needed

Reply:
Ah I see, I've taken a look at replacing the property of name to resource name and it made a few things simpler: no need to cast things as resource name etc. Small change in the tests though